github/open_close_pr.py

```diff
@@ -36,49 +45,178 @@ def load_yaml(yml_path):
         self.repo = github.repository(repo['owner'], repo['name'])
         self.upstream_repo = github.repository('freeipa', 'freeipa')
+        self.args = args
+
+    def get_newest_templ_ver(self):
+        atlas_conf = load_yaml(self.args.atlas_config)
+        token = atlas_conf['token']
+        bearer = 'Bearer {}'.format(token)
+
+        templ_name = TEMPL_NAME.format(
+            flow=self.args.flow,
+            branch=self.args.branch,
+            fedora_version=self.args.fedora_ver
+        )
+
+        url = urljoin(VAGRANT_LINK, templ_name)
+
+        headers = {'Content-Type': 'application/json',
+                   'Authorization': bearer}
+
+        res = requests.get(url, headers=headers)
+        res.raise_for_status()
+
+        latest_ver = res.json()['current_version']['version']
+        logger.info('Latest %s Vagrant box version is: %s', (
+            self.args.branch, latest_ver))
+
+        return latest_ver
+
+    def clean_template_env(self):
+        """
+        We have to clean everything in case there was previously failed
+        create template task
+        """
+        templ_name = TEMPL_NAME.format(
+            flow=self.args.flow,
+            branch=self.args.branch,
+            fedora_version=self.args.fedora_ver
+        )
+        templ_vm_name = templ_name + '_template'
+        box_name = 'f{}'.format(self.args.fedora_ver)
+        libvirt_vol_name = '{}.img'.format(templ_vm_name)
+
+        subprocess.run(['virsh', 'destroy', templ_vm_name])
+        subprocess.run(['virsh', 'undefine', templ_vm_name])
+        subprocess.run(['vagrant', 'box', 'remove', box_name])
+        subprocess.run(['virsh', 'vol-delete', '--pool', 'default',
+                        libvirt_vol_name])
+        try:
+            shutil.rmtree(os.path.join('/tmp', templ_name))
+        except OSError:
+            pass
+
+    def build_new_template(self):
+        inv_path = 'ansible/hosts/runner_localhost'
+        playbook_path = 'ansible/create_template_box.yml'
 
-    def commit_new_prci_config_file(self, new_prci_config, args):
+        os.chmod(os.path.join(self.args.prci_repo_path,
+                              'keys/vagrant'), 0o0600)
+        os.chmod(os.path.join(self.args.prci_repo_path,
+                              'keys/freeipa_pr_ci_insecure'), 0o0600)
+
+        try:
+            logger.info('Started playbook %s', playbook_path)
+            res = subprocess.check_output([
+                    'ansible-playbook',
+                    '-i', inv_path,
+                    playbook_path,
+                    '-e', 'fedora_version={}'.format(self.args.fedora_ver),
+                    '-e', 'git_branch={}'.format(self.args.branch),
+                ], cwd=self.args.prci_repo_path)
+            return res.decode()
+        except subprocess.CalledProcessError:
+            logger.error('Ansible command failed with: %s', res.decode())
+
+    def commit_new_prci_config_file(self, new_template_version):
```

Comment:
`new_template_version` is not used in this method

Reply:
I delete it!